### ISSUE TITLE
Mask for attention

### ISSUE THREAD

**Issue by MordehayM** [SWivid/F5-TTS#819]
### Checks

- [x] This template is only for question, not feature requests or bug reports.
- [x] I have thoroughly reviewed the project documentation and read the related paper(s).
- [x] I have searched for existing issues, including closed ones, no similar questions.
- [x] I confirm that I am using English to submit this report in order to facilitate communication.

### Question details

According to the doc the function scaled get as input the attn with shape [..., L, S] (https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html)
So suppose we have mask like this:
[True, True, True, False], I would like the atn_mask here:
https://github.com/SWivid/F5-TTS/blob/09b478b7d73b54f1c5e5e90f5372126e6d128a74/src/f5_tts/model/modules.py#L432
to be like the following:
[[True, True, True, False],
 [True, True, True, False],
 [True, True, True, False],
 [False, False, False, False]]
by the existing code yields the following atnn_mask:
[[True, True, True, False],
 [True, True, True, False],
 [True, True, True, False],
 [True, True, True, False]]
which is wrong because I don't want to calc the attention in the last query

Please help me understand what I am missing here
Thanks

**Comment by SWivid** [comment]
> to be like the following:
[[True, True, True, False],
[True, True, True, False],
[True, True, True, False],
[False, False, False, False]]

If the attn is like that, why not just seq_len-1?
Read more and distinguish padding mask and attn mask

**Comment by SWivid** [comment]
If confused, try to do the the attention calc process by hand and see how mask function
